### DISCUSSION TITLE
Group properties use composite

### DISCUSSION THREAD

**Discussion by dangngocthanhh** [GrapesJS/grapesjs#5802] [Q&A]
**Hello**

I'm trying to use the "composite" type to combine 4 attributes: top, left, bottom, right into one group. Here's what I've done:

![Screenshot from 2024-04-11 14-59-36](https://github.com/GrapesJS/grapesjs/assets/163803176/6a4e17bc-67ed-49b6-b4db-accd15c6d891)

Along with some custom steps in StylePropertyField (I simulated @artf's code on https://stackblitz.com/edit/grapesjs-react-custom-ui?file=src%2FApp.tsx and then customized it), the final result I achieved is:

![Screenshot from 2024-04-11 14-57-55](https://github.com/GrapesJS/grapesjs/assets/163803176/abb0607b-1140-4979-9bb8-c6d2d6dce03a)

But the issue is when using prop.upValue(value), it cannot be updated. It seems the problem lies in the fact that the inputs are children of the composite but don't have a property for the parent.

Do you have any suggestions for updating the value, or any other ideas? Thanks everyone.

**Comment by artf** [comment]
Please provide a reproducible demo instead of screenshotting the code

**Reply by dangngocthanhh** [reply]
@artf  Thank you for your concern about my issue. I have found my solution. I used CSS order to change the position of the input and kept the default attribute configuration, so the issue has been resolved. For those who encounter this issue, I have created a similar snippet on StackBlitz. Now I will close this discussion.

https://grapesjs-react-custom-ui-xcnjxk.stackblitz.io
